rse-cmdline-git.Rmd

```diff
@@ -0,0 +1,2296 @@
+# Git at the command line {#rse-cmdline-git}
+
+## Questions {#rse-cmdline-git-questions}
+
+```{r, child="questions/rse-cmdline-git.md"}
+```
+
+## Introduction {#rse-cmdline-git-intro}
+
+In a previous lesson (TODO: provide link),
+we learned the basics of Git using a graphical interface (GitKraken or R Studio).
+Like many computer programs,
+using Git at the command line is a potentially more powerful option,
+as it allows you to access all the features.
+This chapter describes how the basic "add, commit, push, pull" cycle works at the command line,
+before introducing new concepts including branching and pull requests.
+
+
+## Setting up Git {#rse-cmdline-git-setup}
+
+When we use Git at the command line for the first time,
+we need to configure a few things. 
+
+On a command line, Git commands are written as `git verb options`,
+where `verb` is what we actually want to do
+and `options` is additional optional information which may or may not be required for the `verb`.
+Here is how Dracula sets up his new laptop:
+
+```shell
+$ git config --global user.name "Vlad Dracula"
+$ git config --global user.email "vlad@tran.sylvan.ia"
+```
+
+(Please use your own name and email address instead of Dracula's.)
+
+For these lessons we will be interacting with [GitHub](https://github.com/),
+so the email address should be the same as the one used when setting up your GitHub account.
+
+The commands we just ran above only need to be run once: the flag `--global` tells Git
+to use the settings for every project, in your user account, on this computer.
+You can re-run them at any time if you want to change your details.
+
+You can also check your settings at any time:
+
+```shell
+$ git config --list
+```
+
+> **Git Help and Manual**
+>
+> Always remember that if you forget a `git` command,
+> you can access the list of commands by using `-h` and access the Git manual by using `--help` :
+>
+> ```shell
+> $ git config -h
+> $ git config --help
+> ```
+
+
+## Creating a repository {#rse-cmdline-git-repos}
+
+Once Git is configured,
+we can start using it.
+
+We will continue with the story of Wolfman and Dracula who are investigating if it
+is possible to send a planetary lander to Mars. 
+
+*TODO: Figure - Motivating Example [../fig/motivatingexample.png](https://github.com/swcarpentry/git-novice/blob/gh-pages/fig/motivatingexample.png)*
+
+First, let's create a directory in `Desktop` folder for our work and then move into that directory:
+
+```shell
+$ cd ~/Desktop
+$ mkdir planets
+$ cd planets
+```
+
+Then we tell Git to make `planets` a repository — a place where Git can store versions of our files:
+
+```shell
+$ git init
+```
+
+It is important to note that `git init` will create a repository that
```

Comment:
I feel that the learners should already know this stuff and that it will be covered in the novice section. I think we could cut this part up to the "git status"

Reply:
Though, I guess it could be kept, especially if in the novice we focus on the GUI interaction with Git while this is the commandline.

Reply:
Maybe add a bit near the beginning of this section along the lines of: 
"This section covers how to generate a git repository, associated with a project folder. If you already know how to do this at the command line, skip to the next section." 

I think we should have a "topic sentence" regardless, so I'll suggest that above.